### PR TITLE
Skip all but first test of each example

### LINKED ISSUE #151: Skipping tests?
Would there be any interest in skipping later tests in exercises? I'm thinking something similar to `xit` used in [the Ecmascript exercises](https://github.com/exercism/ecmascript/blob/master/exercises/accumulate/accumulate.spec.js). Syntactically it would be a little more heavy-weight in Elm, so I've opted to ask in an issue before making the change for pull request.

Also related, I noticed what seems to be an error in the elm-test README about this feature. See https://github.com/elm-community/elm-test/pull/186.

### PR DESCRIPTION
**Pull request by iamvery**
This allows learners to gradually approach exercise. However, unlike commented tests, the Elm compiler is still able to infer type information from the skipped tests.

Note: I made this change very mechanically, it's quite possible that the very first test is not the ideal starting point for learners. I'd really appreciate some review of the changes here before merge.

Thank you! ❤️ 

[Closes #151]

### CODE REVIEW

**Comment by tgecho** [comment]
So I thought of a problem a few hours ago, and Travis is backing me up with the [test failure](https://travis-ci.org/exercism/elm/builds/250512945?utm_source=github_status&utm_medium=notification): We still need to run all of the tests against the examples in CI. Unfortunately elm-test doesn't have a way to "force" running incomplete tests.

The best (but still bad) I can come up with offhand is to temporarily remove the skips (`sed 's/skip <\|//g'`) when we do the CI build. It's a bit gross, but I think optimizing for the end users is important enough here.

I have this commit prepared. Just doing another skim through the rest and I'll merge the lot.

File: exercises/nucleotide-count/tests/Tests.elm [hunk 1]
```diff
@@ -10,16 +10,19 @@ tests =
     describe "NucleotideCount"
         [ test "the solution is for the correct version of the test" <|
             \() -> Expect.equal 2 version
```
Comment (tgecho) [inline exercises/nucleotide-count/tests/Tests.elm]: We should probably unskip the first actual test. The version check should automatically pass :)

File: exercises/run-length-encoding/tests/Tests.elm [hunk 2]
```diff
@@ -10,28 +10,36 @@ tests =
...
-            \() ->
-                Expect.equal "WWWWWWWWWW"
-                    (decode "10W")
-        , test "encode unicode" <|
-            \() -> Expect.equal "⏰3⚽2⭐⏰" (encode "⏰⚽⚽⚽⭐⭐⏰")
-        , test "decode unicode" <|
-            \() -> Expect.equal "⏰⚽⚽⚽⭐⭐⏰" (decode "⏰3⚽2⭐⏰")
+        , skip <|
+            test "encode simple" <|
+                \() -> Expect.equal "2A3B4C" (encode "AABBBCCCC")
```
Comment (tgecho) [inline exercises/run-length-encoding/tests/Tests.elm]: Another first actual test 🔍

File: exercises/sublist/tests/Tests.elm [hunk 3]
```diff
@@ -10,40 +10,58 @@ tests =
...
-            \() -> Expect.equal Superlist (sublist [ 1, 1, 1, 2 ] [ 1, 1, 2 ])
-        , test "superlist early in huge list" <|
-            \() -> Expect.equal Superlist (sublist (List.range 1 100000) [ 3, 4, 5 ])
-        , test "recurring values sublist" <|
-            \() -> Expect.equal Sublist (sublist [ 1, 2, 1, 2, 3 ] [ 1, 2, 3, 1, 2, 1, 2, 3, 2, 1 ])
-        , test "recurring values unequal" <|
-            \() -> Expect.equal Unequal (sublist [ 1, 2, 1, 2, 3 ] [ 1, 2, 3, 1, 2, 3, 2, 3, 2, 1 ])
+        , skip <|
+            test "empty equals empty" <|
+                \() -> Expect.equal Equal (sublist [] [])
```
Comment (tgecho) [inline exercises/sublist/tests/Tests.elm]: Another first actual test 🔎

File: exercises/robot-simulator/tests/Tests.elm [hunk 4]
```diff
@@ -15,46 +15,51 @@ tests =
...
+                        \() -> Expect.equal South robot.bearing
                 ]
             )
         , describe "turn right"
             ((List.range 1 3)
                 |> List.scanl (\_ r -> turnRight r) defaultRobot
                 |> List.map .bearing
                 |> assertionList [ North, East, South, West ]
-                |> List.indexedMap (\i e -> test ("step " ++ toString i) (\() -> e))
+                |> List.indexedMap (\i e -> skip <| test ("step " ++ toString i) (\() -> e))
```
Comment (tgecho) [inline exercises/robot-simulator/tests/Tests.elm]: I'm thinking this skip might be better at the describe level. elm-test won't let them forget it, but I think it'll be a bit more clear there. Also see the next describe block.

File: exercises/nucleotide-count/tests/Tests.elm (hunk 1 above)
Comment (iamvery) [inline exercises/nucleotide-count/tests/Tests.elm]: I had noticed the version function on another exercise, but I don't think there was a test for it. What's the story with that?

File: exercises/robot-simulator/tests/Tests.elm (hunk 4 above)
Comment (iamvery) [inline exercises/robot-simulator/tests/Tests.elm]: That makes a ton of sense! Agree 👍

**Comment by iamvery** [comment]
I didn't realize you could "force" any test runners to ignore skips! Do those options exist for exactly this sort of use case? Would it be a worthwhile addition to elm-test?

Is there anything else you would like me to do here, or are you taking over?

File: exercises/nucleotide-count/tests/Tests.elm (hunk 1 above)
Comment (tgecho) [inline exercises/nucleotide-count/tests/Tests.elm]: Sometimes we change a test in a way that isn't backwards compatible. The version declaration in the exercise stub helps reviewers know what tests they were writing against. That's the idea at least... we haven't been perfectly consistent.

**Comment by tgecho** [comment]
The only one I know of is nose (an old python test runner). Not too many use cases that I can think of, since IMO skips are usually temporary or a smell.

I have a commit prepped to do the skipping. If you don't have time to do the adjustments above I can take care of them in the next day or so to get this merged.

**Comment by iamvery** [comment]
I believe I've added all the changes you requested. Thanks for your time and feedback :)

**Comment by tgecho** [comment]
Thanks for your help! I think this is a really nice polishing up of the experience!

**Comment by iamvery** [comment]
Thanks for the assists! ❤️

**Pull request merged**